### ISSUE TITLE
lyra2z330. speed is 0 H/s more than 5 minutes

### ISSUE THREAD

**Issue by Quake4** [JayDDee/cpuminer-opt#189]
![image](https://user-images.githubusercontent.com/29517243/59923376-f7a0c280-943b-11e9-8c8c-14ef1a4d9ffc.png)

**Comment by Quake4** [comment]
![image](https://user-images.githubusercontent.com/29517243/59924327-2f106e80-943e-11e9-81b0-bbbedb9c685a.png)

**Comment by Quake4** [comment]
other algos show speed after 15 secs (in api)

**Comment by JayDDee** [comment]
The hash rate reported by the miner is artificial so you aren't losing anything. The important part is
the share difficulty. This represents the true value of each share.

I made an optimization to tighten the scan loop. Previously when a share was found the scan loop
would exit and the hash rate would be counted and some other housekeeping before starting a
new scan.

The the scan continues until the pool sends a new job or after a certain period of time. The gain
is the CPU starts the next hash sooner. The gain is larger for high hash rate algos and lower
for low hash rate algos like lyra2z330 because the overhead represents a different proportion
of the work to calculate one hash (one loop interation).

The side effect is the hash rate isn't known until the first scan ends.

In short the hash rate displayed is for the previous scan, not the current one. On the first scan
there is no previous scan therefore no hash rate. All shares submitted on the first scan will
show zero hash rate. You still get full value for each share regardless of the reported hash rate.

I'm looking at a completely different way to calculate hash rate using the info from the pool
instead of counting loops in the code. In the meantime you'll have to take my word that it's
actually better now because the miner doesn't spend as much time counting and reporting,
and more time hashing.

I can make the hashrate say anything you want but it doesn't mean anything.

**Comment by Quake4** [comment]
Thank you for full relay.

**Comment by JayDDee** [comment]
No problem. I should add this only happens for algos that report thread and/or lane info in the
"Share submitted" message. In the next release more algos will do it as I continue the restructuring.
